pkg/cmd/cli/cmd/set/imagelookup.go

```diff
@@ -0,0 +1,235 @@
+	cmdNamespace, explicit, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	clientConfig, err := f.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	o.OutputVersion, err = kcmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+	if err != nil {
+		return err
+	}
+
+	o.PrintTable = len(args) == 0 && !o.All
+
+	mapper, typer := f.Object()
+	o.Builder = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(explicit, &resource.FilenameOptions{Recursive: false, Filenames: o.Filenames}).
+		SelectorParam(o.Selector).
+		SelectAllParam(o.All).
+		ResourceNames("imagestreams", args...).
+		Flatten()
+	if len(args) == 0 {
+		o.Builder.SelectAllParam(true)
+		o.Builder.ResourceTypes("imagestreams")
+	}
+
+	output := kcmdutil.GetFlagString(cmd, "output")
+	if len(output) != 0 {
+		o.PrintObject = func(obj runtime.Object) error { return f.PrintObject(cmd, mapper, obj, o.Out) }
+	}
+
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = kcmdutil.GetFlagString(cmd, "output") == "name"
+	o.Mapper = mapper
+
+	return nil
+}
+
+// Validate verifies the provided options are valid or returns an error.
+func (o *ImageLookupOptions) Validate() error {
+	return nil
+}
+
+// Run executes the ImageLookupOptions or returns an error.
+func (o *ImageLookupOptions) Run() error {
+	infos := o.Infos
+	singleItemImplied := len(o.Infos) <= 1
+	if o.Builder != nil {
+		loaded, err := o.Builder.Do().IntoSingleItemImplied(&singleItemImplied).Infos()
+		if err != nil {
+			return err
+		}
+		infos = loaded
+	}
+
+	if o.PrintTable && o.PrintObject == nil {
+		return o.printImageLookup(infos)
+	}
+
+	patches := CalculatePatches(infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		info.Object.(*imageapi.ImageStream).Spec.ResolveLocalNames = o.Enabled
+		return true, nil
+	})
+	if singleItemImplied && len(patches) == 0 {
+		return fmt.Errorf("%s no changes", infos[0].Name)
+	}
+	if o.PrintObject != nil {
+		object, err := resource.AsVersionedObject(infos, !singleItemImplied, o.OutputVersion, kapi.Codecs.LegacyCodec(o.OutputVersion))
+		if err != nil {
+			return err
+		}
+		return o.PrintObject(object)
+	}
+
+	failed := false
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			failed = true
+			fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+			continue
+		}
+
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			fmt.Fprintf(o.Err, "info: %s %q was not changed\n", info.Mapping.Resource, info.Name)
+			continue
+		}
+
+		glog.V(4).Infof("Calculated patch %s", patch.Patch)
+
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, kapi.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			handlePodUpdateError(o.Err, err, "altered")
```

Comment:
Can you rename that method to something more generic, if you're using it here. It is very confusing in this context with the current name.